internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -889,12 +902,16 @@ public CompletableFuture<PubSubProduceResult> put(
     byte[] serializedKey = keySerializer.serialize(topicName, key);
     byte[] serializedValue = valueSerializer.serialize(topicName, value);
     int partition = getPartition(serializedKey);
-
     int replicationMetadataPayloadSize = putMetadata == null ? 0 : putMetadata.getSerializedSize();
     isChunkingFlagInvoked = true;
-    if (serializedKey.length + serializedValue.length
-        + replicationMetadataPayloadSize > maxSizeForUserPayloadPerMessageInBytes) {
-      if (isChunkingEnabled) {
+
+    /**
+     * {@link RecordTooLargeException} will be thrown unless the record size fits within one of the following categories:
+     * Records without Chunking < ~1MB < Records with Chunking < MAX_RECORD_SIZE_BYTES
```

Comment:
This comment is confusing. The `~1MB < Records with Chunking < MAX_RECORD_SIZE_BYTES` is not always true. Records in stores with chunking can be smaller than 1 MB too.

Reply:
Changed to `Chunking Not Needed < ~1MB < Chunking Needed < MAX_RECORD_SIZE_BYTES`.

By `Records with Chunking`, the chunking is referring to records that are already chunked rather than pertaining to whether chunking is enabled on the store, but maybe the new comment is more clear?